buildpack/bp_descriptor.go

```diff
@@ -83,7 +86,7 @@ func ReadBpDescriptor(path string) (*BpDescriptor, error) {
 					descriptor.Targets = append(descriptor.Targets, TargetMetadata{TargetPartial: TargetPartial{OS: "windows", Arch: "amd64"}})
 				}
 				if fname == "build" {
-					descriptor.Targets = append(descriptor.Targets, TargetMetadata{TargetPartial: TargetPartial{OS: "linux", Arch: "amd64"}})
+					descriptor.Targets = append(descriptor.Targets, TargetMetadata{TargetPartial: TargetPartial{OS: "linux", Arch: "amd64"}}) // Question: why is the arch part of this inference -- should we leave the arch empty when we infer the OS?
```

Comment:
That's a good question - I don't think we should set anything besides the `OS` if we can. Otherwise a buildpack that could work across arm/amd64 would not work on an arm image unless they specifically added the targets. Right?

Reply:
Thanks - that's exactly what i mean!   I think... ok, here's the tension. 

[our spec](https://github.com/buildpacks/spec/pull/335/files#diff-e603760990971da3f77be4bb8d77c3405098f006814fd8c054d2d15f395b8330R193) says we should abide by the OCI spec .  

the OCI spec says that arch values SHOULD conform to valid GOARCH values, 

which are:
> Choices for $GOARCH are amd64 (64-bit x86, the most mature port), 386 (32-bit x86), arm (32-bit ARM), arm64 (64-bit ARM), ppc64le (PowerPC 64-bit, little-endian), ppc64 (PowerPC 64-bit, big-endian), mips64le (MIPS 64-bit, little-endian), mips64 (MIPS 64-bit, big-endian), mipsle (MIPS 32-bit, little-endian), mips (MIPS 32-bit, big-endian), s390x (IBM System z 64-bit, big-endian), and wasm (WebAssembly 32-bit).


hmmm so i guess we're already diverging a little bit here by adding the handling of `*` stacks - honestly this might be another reasonable place to put a `*` though - in our case these could be e.g. just shell scripts, which would be expected to execute on any architecture,  right?  so it feels reasonable to keep using a wildcard,  unless that means we're going to eventually output something that some OCI registry will refuse to accept?    but the alternative would be that buildpack authors would have to maintain two identical things that differ only by specifying arm/x86?

Like we're missing a generic value for `arch` (or "generics" for the DSL describing arch) because it's a non-sequitor from the perspective of the go compiler, and we're a few levels deep into people saying "let's just use whatever is working for the layer below us" ,  right?

Reply:
Yeah, I think I like wildcard myself. The bash/executable that runs everywhere is far more likely for our project. 🤷

Reply:
cool - updated here and left a comment in the spec PR: https://github.com/buildpacks/spec/pull/335/files#r1137876951